PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfigLoader.java

```diff
@@ -100,7 +100,7 @@ private Map<String, RuleConfig> loadRuleConfigurations(String filePath) {
             // If not, throw an exception
             // This is to ensure that the JSON file is in the correct format
             if (reader.nextToken() != JsonToken.START_OBJECT) {
-                throw new IOException("Expected start of object");
+                throw new IllegalArgumentException("Expected start of object");
```

Comment:
What happens when this exception is thrown? Will this crash the IDE? Will this disable the plugin?

Reply:
nope -- there's a skipRuleCheck() method returning a boolean s if a rule doesn't have its configs set up the check is automatically skipped so other rules in the plugin will still work as normal